src/core/scene/a-scene.js

```diff
@@ -403,13 +461,28 @@ module.exports = registerElement('a-scene', {
     render: {
       value: function (time) {
         var effect = this.effect;
+        var camera = this.camera;
         var timeDelta = time - this.time;
 
         if (this.isPlaying) { this.tick(time, timeDelta); }
 
-        this.animationFrameID = effect.requestAnimationFrame(this.render);
-        effect.render(this.object3D, this.camera);
-        this.time = time;
+        window.performance.mark('render-iteration-started');
+        if (this.getAttribute('post-processing')) {
+          var size = this.renderer.getSize();
+          var renderTarget = this.renderTarget;
+          if (size.width !== renderTarget.width || size.height !== renderTarget.height) {
+            renderTarget.setSize(size.width, size.height);
```

Comment:
It does a check first so setSize won't actually happen every frame. The reason I went that route, checking every frame, was because VREffect will switch the framebuffer(canvas) size internally to fit the hmd without providing any feedback